source/bayes_simulation.Rmd

```diff
@@ -45,112 +45,171 @@ correctly identifies 80 percent as "OK" but says that 20 percent are
-sort about which you are presently inquiring.
+dice = np.random.random(N)
+mechanic_says_OK = np.zeros(N, dtype=bool)
+mechanic_says_OK[faulty & (dice < 0.1)] = True
+mechanic_says_OK[~faulty & (dice < 0.8)] = True
 
-Using RESAMPLING STATS, an answer may be produced as follows:
+k = np.count_nonzero(mechanic_says_OK & faulty) / np.count_nonzero(mechanic_says_OK)
 
-"01 - 27" = actually faulty, said to be "faulty" "28 - 30 = faulty, said
-to be "OK"
+print(f'{k * 100:.2}%')
+```
 
-"31 - 86" = OK, "OK"
+The result is around 5.1%.  If we look at the numbers on our diagram, this makes sense: if we started with 100 cars, 59 are declared OK by a mechanic.  Of those, 3 were faulty, and `3/59 = 0.0508`.
 
-"87 - 100" = OK, "faulty"
+### Probability interpretation
 
-`           REPEAT 1000         `
+**Probability from proportion**
 
-Do 1000 repetitions
+In these examples, it is very common for us to calculate proportions.  In a given simulation: How many cars are declared OK by a mechanic?  59/100.  How many of those 59 were faulty?  3/59.  Given evidence of events, we can count how common they were.
 
-`           GENERATE 1 1,100 a         `
+A natural extension of this idea is to predict the probability of an event occurring.  We can say "in past 100 experiments, condition X occurred 20 times, therefore I  estimate that the probability of X occurring, $P(X)$, is 20/100".  The more trial outcomes we examine, the better our probability estimates become.
 
-Generate a number between "1" and "100"
+**Ratios of proportions / probabilities**
 
-`           IF a between 28 86         `
+At our mechanic's yard, we ask "how many red cars here are broken"?  To calculate that, we first count the number of red cars, then the number of those red cars that are broken, then calculate the ratio: `red_cars_broken / red_cars`.
 
-If it's between "28" and "86" (those that say "good")
+We could just as well have worked in percentages, and we would have gotten the same answer: `percentage_of_red_cars_broken / percentage_of_cars_that_are_red`, since
+that is `(red_cars_broken / 100) / (red_cars / 100)`.  And probabilities would have worked too, given our assumption that the counts are good representations of the yard: `P(red_cars_broken_in_the_yard) / P(red_cars_in_the_yard)`.
 
-`           SCORE a z         `
+**Probability relationships: conditional probability**
 
-Score this number
+We return to the quantity computed above: the **number** of cars being faulty *given* that the mechanic said they were OK.  We use this ratio as an estimate of the probability that that will be true, i.e: the **probability** of cars being faulty *given* that the mechanic said they were OK.
 
-`           END         `
+This is called a *conditional probability*, $P(X | Y)$.  Given a certain outcome Y, calculate the probability of X occurring.
 
-End the IF condition
+Conditional probabilities are calculated as follows:
```

Comment:
Here is where I'd go to my Venn diagram to show why that is the right calculation.  Or maybe we need a simulation and then the Venn diagram to show why.

Reply:
I guess this should all go in the section on conditional probabilities.